docs/DeveloperGuide.md

```diff
@@ -283,31 +283,112 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### Use cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is the `TutorAid` and the **Actor** is the `Tutor`, unless specified otherwise)
 
-**Use case: Delete a person**
+**Use case 4: Update a student's progress**
 
 **MSS**
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1. Tutor views the list of his/her students
+2. TutorAid displays the list of students
+3. Tutor identifies the student index to update their progress
+4. Tutor updates progress for that student
+5. TutorAid registers the new progress for the student at the specified index number
 
     Use case ends.
 
 **Extensions**
 
-* 2a. The list is empty.
+* 1a. Invalid list command syntax
+
+    * 1a1. TutorAid displays an error message and requests the tutor to re-enter the command.
+
+    * 1a2. Tutor re-enters the command.
+
+      Steps 1a1-1a2 are repeated until the syntax entered is correct. Use case resumes from step 2.
+
+* 1b. The list is empty.
 
   Use case ends.
 
-* 3a. The given index is invalid.
+* 4a. Invalid add/delete progress syntax
+
+    * 4a1. TutorAid displays an error message and requests the tutor to re-enter the command.
+
+    * 4a2. Tutor re-enters the command
+
+      Steps 3a1 - 3a2 are repeated until a valid syntax is given. Use case resumes from step 5
+  
+* 5a. Incorrectly updating the progress of another student
+
+    * 5a1. Tutor uses add/delete progress commands to revert the incorrect changes. Use case resume from step 3
+
+**Use case 7: Add a todo list for a session**
+
+**MSS**
+
+1. Tutor views his schedule
+2. TutorAid displays the schedule
+3. Tutor identifies the session to add a todo list
+4. Tutor adds a todo task to that session
+5. TutorAid registers the task to the session
+6. Tutor repeats from step 2 until he/she finishes adding all the tasks
```

Comment:
Since the use case focuses on the tutor only adding todo tasks for 1 session, maybe the tutor can just repeat step 4 for every subsequent task?

Reply:
I agree with this, since it may incur extra unnecessary work for the tutor to repeat step 3 in order to add a todo task again. Otherwise, LGTM.